src/neptune/internal/utils/dependency_tracking.py

```diff
@@ -32,7 +32,9 @@ def log_dependencies(self, run: "Run") -> None:
 class InferDependenciesStrategy(DependencyTrackingStrategy):
     def log_dependencies(self, run: "Run") -> None:
         try:
-            dependencies_str = subprocess.check_output([sys.executable, "-m", "pip", "freeze"]).decode("utf-8")
+            dependencies_str = subprocess.check_output([sys.executable, "-m", "pip", "list", "--format=freeze"]).decode(
```

Comment:
Have you tried the `importlib` approach? I've been suggesting something like:
```python
import sys

if sys.version_info >= (3, 8):
    from importlib.metadata import distributions
else:
    from importlib_metadata import distributions


def list_installed_packages():
    dists = list(sorted(distributions(), key=lambda d: d.metadata['Name']))

    for dist in dists:
        name, version = dist.metadata['Name'], dist.metadata['Version']
        print(f'{name}=={version}')


def main():
    list_installed_packages()


if __name__ == '__main__':
    main()
```

Reply:
@SiddhantSadangi

Reply:
Just tried. This seems to work very well!

Lets do this instead of subprocess